src/core/linter-rules/local-refs-exist.js

```diff
@@ -21,9 +21,22 @@ export function run(conf) {
     return;
   }
 
+  /**
+   * Skip links without hashes; and those that point to outside current page.
+   * @returns
+   */
+  const shouldTest = (() => {
+    const { origin, pathname } = new URL(document.baseURI);
+    /** @type {(a: HTMLAnchorElement) => boolean} */
+    const fn = a => a.hash && a.origin === origin && a.pathname === pathname;
+    return fn;
+  })();
```

Comment:
Might be more consistent just to make is a regular function above `isBrokenHyperlink()` ... can still curried , but just maybe just:

 ```suggestion
  const shouldTest = checkHashSameOrigin();
```

Or even:
```
   .filter(checkHashSameOrigin())
```